File: 7-kyu/drone-fly-by/c++/solution.cpp
std::string flyBy(std::string lamp, std::string drone)
{
    int drone_pos = -1;
    for (unsigned i = 0; i < drone.length(); i++) {
        if (drone[i] == 'T') {
            drone_pos = i;
            break;
        }
    }
    if (drone_pos > -1) {
        for (unsigned i = 0; i <= drone_pos; i++) {
            lamp[i] = 'o';
        }
    }
    return lamp;
}

File: 7-kyu/drone-fly-by/swift/solution.swift
func flyBy(lamps: String, drone: String) -> String {
    let lampsCount = lamps.count
    let droneCount = drone.count
    let oRepeatCount = droneCount > lampsCount ? droneCount - 1 : droneCount
    let xRepeatCount = max(min(lampsCount - droneCount, lampsCount), 0)
    return "\(String(repeating: "o", count: oRepeatCount))\(String(repeating: "x", count: xRepeatCount))"
}